### ISSUE TITLE
[Docs] Pool execute or query is automatically create and release a connection

### ISSUE THREAD

**Issue by mems** [sidorares/node-mysql2#745]
The docs should mention that methods of connection pool `execute()` and `query()` internally create and release a connection each time it called:

```
const pool = await mysql.createPool({/*...*/});
await = pool.execute(query);
// is same as
const conn = await pool.getConnection();
await conn.execute(query);
conn.release();
```

**Comment by sidorares** [comment]
yes, what's your question @mems ?

**Comment by mems** [comment]
Just to add that clarification in docs

**Comment by sidorares** [comment]
agreed, just had a look at `/documentation/Promise-Wrapper.md` and we should probably have good detailed api documentation there and not just couple of brief examples

**Comment by soorajvnair** [comment]
Yes this is important, because i've seen people manually accquire and release connection when working with query and execute, which is not necessary as @mems has highlighted.

**Comment by anthgur** [comment]
I agree with this. I went from using `execute` (implicit connection) to using transactions (user managed connection) and I got bit by a pool deadlock.

**Comment by frenzymind** [comment]
what bout if I use 'mysql2/promise'  ?
connection = await mysql.createConnection({ ... })
const [rows] = await connection.execute( ...  )
Should I call connection.end() after ?
